src/instruments/src/Common/displayUnit.tsx

```diff
@@ -39,6 +42,27 @@ export const DisplayUnit: React.FC<DisplayUnitProps> = (props) => {
         }
     });
 
+    // TODO: This only calculates in one instance (on Captain's PFD), move this to FCU when in react.
+    if (props.potentiometerIndex === 88) {
+        const [zoomLevel] = useSimVar('COCKPIT CAMERA ZOOM', 'percent', 200);
+        const camXyz = useGameVar('CAMERA POS IN PLANE', 'xyz', 200);
+
+        useEffect(() => {
+            // zTarget: Target zPos that marks max pixel effect point (11.625z @ 100%, 11.7z @ 75%)
+            const zTarget = (3975 - zoomLevel) / 333.33;
+            // zΔ: Diff between current zPos and zTarget
+            const zDelta = camXyz.z - zTarget;
+            // opacity: 4zΔ + 0.5 < 0.5
+            setOpacity(Math.min(0.5, 4 * (zDelta) + 0.5));
+            setSimVar('L:A32NX_LCD_MASK_OPACITY', opacity, 'number');
+        }, [camXyz, zoomLevel]);
+    } else {
+        const [maskOpacity] = useSimVar('L:A32NX_LCD_MASK_OPACITY', 'number');
+        useEffect(() => {
+            setOpacity(maskOpacity);
+        }, [maskOpacity]);
+    }
```

Comment:
I know this PR is still in draft, but I'm afraid I'll forget otherwise.

**issue**: Encapsulation breaks down.
**suggestion**: Move the things that do not directly relate to this component into their own component. This means `DisplayUnit` will only contain the code reading the SimVar and setting the opacity. Not the code calculating the opacity and writing it to a SimVar.
**suggestion**: Decide whether or not to render the newly created component within the parent, such that you don't need to check for equality to potentiometer 88 (which is only true for this aircraft/for now); but instead can check `isCaptainSide()`.
**suggestion**: Avoid using `setSimVar` in a React context. Use `useSimVar`.

Reply:
Hopefully I can learn from this question and I hope it doesn’t sound stupid, but what is the point of separation of logic out of displayUnit if the logic immediately ties into its functionality? Wouldn’t further abstraction just result in more clutter?

Reply:
Feel free to ask questions. I'll happily explain my thoughts.

## Potentiometer 88
When thinking of components, think of them as a tree. Each item in the tree is allowed to know (1) about itself, and (2) about its children. A parent can provide a child with information so that it can make decisions (as is done by passing the `potentiometerIndex`), but the child shouldn't have knowledge about the implementation of the parent.

The fact that potentiometer 88 is the captain side PFD is a fact only known to the `PFD` component. Imagine we change the logic in the `PFD`. We will then also have to change the logic here. If we forget, then things will break (and we might not notice). We can prevent this problem by not depending on knowing 88 = captain PFD at all.

## Remaining implementation
It seems we only want to execute the calculation once. Instead of using the potentiometer we could pass a boolean indicating if the calculation should take place in this particular `DisplayUnit`. However, we shouldn't, as we can't reasonably argue that a single `DisplayUnit` should have the responsibility of calculating the opacity for all display units, as we have to remember that the `DisplayUnit` should be unaware of any other `DisplayUnit`, given that those other `DisplayUnit`s are not below it in the tree.

In fact, moving this up the tree might not be good enough either, as we're still calculating this within a single instance of a PFD.

So there are two choices:
1. Calculate it in every DisplayUnit (which would be my preferred option), or
2. Calculate and write the opacity in `A32NX_Core` and then each `DisplayUnit` would merely read the SimVar.

Reply:
Yes this was just a bit of mess I left in as I was drafting. I haven't decided which approach I want to take on this yet.

Reply:
@2hwk 1 feels more encapsulated than 2. Though both are good options and 2 might be better in the current situation as we're having multiple ways of coding displays (React, old style, etc).

Reply:
> @2hwk 1 feels more encapsulated than 2. Though both are good options and 2 might be better in the current situation as we're having multiple ways of coding displays (React, old style, etc).

Ok, I see your point, I'll probably be moving this logic into A32NX_Core then, I would probably adopt the first stance if we still didn't have to worry about the remaining original javascript, but as it is I think I will leave that work to future contributors.

Reply:
I have a question, why is the logic tied to the potentiometerIndex?

Reply:
That was just a fast and dirty way of restricting the useEffect hook that controls the logic to only ran on the captain's side PFD.